sql-to-dbsp-compiler/SQL-compiler/src/main/java/org/dbsp/sqlCompiler/ir/expression/literal/DBSPLiteral.java

```diff
@@ -135,19 +136,21 @@ public void accept(InnerVisitor visitor) {
         visitor.postorder(this);
     }
 
-    public void checkNotNull() {
-        if (this.isNull)
-            throw new InternalCompilerError("Expected a non-null literal", this);
-    }
-
     boolean hasSameType(DBSPLiteral other) {
         return this.getType().sameType(other.getType());
     }
 
     /**
-     * The same literal but with a non-nullable type.
+     * The same literal but with the specified nullability.
      */
-    public abstract DBSPLiteral getNonNullable();
+    public abstract DBSPLiteral getWithNullable(boolean mayBeNull);
```

Comment:
Ah now I see. I assume maybeNull = true gives an `Option<bool>`  and `bool` otherwise?